pkg/fftm/manager.go

```diff
@@ -79,16 +84,48 @@ type manager struct {
 	enableChangeListener  bool
 }
 
-func NewManager(ctx context.Context) (Manager, error) {
+func InitConfig() {
+	tmconfig.Reset()
+	events.InitDefaults()
+}
+
+func NewManager(ctx context.Context, connector ffcapi.API) (Manager, error) {
 	var err error
+	m := newManager(ctx, connector)
+	if m.name == "" {
+		return nil, i18n.NewError(ctx, tmmsgs.MsgConfigParamNotSet, tmconfig.ManagerName)
+	}
+	m.confirmations = confirmations.NewBlockConfirmationManager(ctx, m.connector)
+	m.policyEngine, err = policyengines.NewPolicyEngine(ctx, tmconfig.PolicyEngineBaseConfig, config.GetString(tmconfig.PolicyEngineName))
+	if err != nil {
+		return nil, err
+	}
+	wsconfig := wsclient.GenerateConfig(tmconfig.FFCoreConfig)
+	m.wsClient, err = wsclient.New(m.ctx, wsconfig, nil, m.startChangeListener)
+	if err != nil {
+		return nil, err
+	}
+	m.apiServer, err = httpserver.NewHTTPServer(ctx, "api", m.router(), m.apiServerDone, tmconfig.APIConfig, tmconfig.CorsConfig)
+	if err != nil {
+		return nil, err
+	}
+	if err = m.initPersistence(ctx); err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+func newManager(ctx context.Context, connector ffcapi.API) *manager {
 	m := &manager{
-		connectorAPI:     ffcapi.NewFFCAPIClient(ctx, tmconfig.ConnectorPrefix, ffcapi.Variant(config.GetString(tmconfig.ConnectorVariant))),
-		ffCoreClient:     ffresty.New(ctx, tmconfig.FFCorePrefix),
+		connector:        connector,
+		ffCoreClient:     ffresty.New(ctx, tmconfig.FFCoreConfig),
```

Comment:
I think we should have a check somewhere in this function to make sure `ffcore.url` is set. (Or maybe in ffresty?) The default value is `nil` and it will get initialized with an empty string without returning an error. This results in a failure later when code attempts to use the manager at runtime, rather than immediately printing an error telling the user the missing config key at startup.

Reply:
This has been removed now, as all remnants of the code that used the FF Core SPI have been removed (in favor of using the internal LevelDB and emitting WS replies)